keyboards/handwired/kbod/keymaps/default/keymap.c

```diff
@@ -53,25 +58,24 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
   ),
 };
 
-const uint16_t PROGMEM fn_actions[] = {
-  [0]  = ACTION_FUNCTION(0),
-  [1]  = ACTION_FUNCTION(1),
-};
-
-
-void action_function(keyrecord_t *record, uint8_t id, uint8_t opt) {
+bool process_record_user(uint16_t keycode, keyrecord_t* record) {
```

Comment:
Does the proposed clang-format file have anything to say about this? Personally I think of the `*` as part of the type, because the behaviour and usage of the argument changes based on its presence whereas `record` is just the name given to it.

Reply:
Yeah, it also suggests pointer alignment on the right (see [here](https://github.com/qmk/qmk_firmware/pull/4880/files#diff-69a68f7ce1ea3687a38d8e4705e6fa48R18)). However, keep in mind that the format file hasn't been approved yet, so for now we have to stick with what the most common usage is in existing firmware code. And that's `keyrecord_t *record` by far: 1576 occurrences vs only 9 for `keyrecord_t* record`.

By the way, your argument breaks down if you want to declare multiple pointer variables. That's why pointers are a derived type and not a regular type. Further, according to the C standard and the language grammar, the `*` _is_ part of the declarator and not the type (see [here](https://en.cppreference.com/w/c/language/declarations#Declarators)). I personally prefer right alignment in C code and left alignment in C++ code.

Reply:
Fair enough. Although, declaring multiple variables on a single line is also a no-no in my book :P Which is why it just looks wrong to me.

Reply:
I agree with you on that one. But I still like to use right alignment in case someone else does it like that.